Add EVM integration to docs (#202)
timewave-computer/valence-protocol: docs/src/authorizations_processors/processor.md

### Execution

When a processor is `Ticked`, the first `Message Batch` will be taken from the queue (`High` if there are batches there or `Med` if there aren’t).
After taking them, we will execute them in different ways depending if the batch is `Atomic` or `NonAtomic`.

- For `Atomic` batches, the `Processor` will execute either all of the functions or none of them. If this execution fails, we will check the `RetryLogic` of the batch to decide if the match is to be re-queued or not. If not, we will send a callback with `Rejected(error)` status to the authorization contract).
  If the execution succeeded we will send a callback with `Executed` status to the Authorization contract.

- For `NonAtomic` batches, we will execute the functions one by one and applying the RetryLogic individually to each function if they fail. `NonAtomic` functions might also be confirmed via `CallbackConfirmations` in which case we will keep them in a separate storage location until we receive that specific callback.
  Each time a function is confirmed, we will re-queue the batch and keep track of what function we have to execute next.

>   If at some point a function uses up all its retries, we will send a callback to the Authorization contract with a `PartiallyExecuted(num_of_functions_executed, execution_error)` status. If all of them succeed it will be `Executed` and if none of them were it will be `Rejected(error)`.

Author: Rephrased for more clarity. If some of them are executed it's PartiallyExecuted(num_of_functions_successfully_executed)

>   For `NonAtomic` batches, we need to tick the processor each time the batch is at the top of the queue to continue, so we will need at least as many ticks as number of functions we have in the batch, and each function has to wait for its turn.

Reviewer: ```suggestion
  For `NonAtomic` batches, the processor must be ticked each time the batch is at the top of the queue to continue, so at least as many ticks will be required as the number of functions in the batch.
```

### Storage

> The `Processor` will receive `Message Batches` from the authorization contract and will enqueue them in a custom storage structure we designed for this purpose, called a `QueueMap`. This structure is a FIFO queue with owner privileges (allows the owner to insert or remove from any position in the queue).

Reviewer: ```suggestion
The `Processor` will receive `Message Batches` from the Authorization contract and will enqueue them in a custom storage structure called a `QueueMap`. This structure is a FIFO queue with owner privileges, which allow the owner to insert or remove messages from any position in the queue.
```
